src/groups.py

```diff
@@ -1,15 +1,27 @@
 
+# all dependancy on nodeattr is to be located in this file
 from plumbum import local
-
-__nodeattr = local['nodeattr']['-f', '/var/lib/adminware/genders']
+from tempfile import NamedTemporaryFile
 
 def list():
-    groups = __nodeattr('-l').split("\n")
+    groups = __nodeattr()('-l').split("\n")
     groups.remove('')
     return groups
 
 def nodes_in(group_name):
-    nodes = __nodeattr('-n', group_name).split("\n")
+    nodes = __nodeattr()('-n', group_name).split("\n")
     nodes.remove('')
     return nodes
 
+def expand_nodes(node_list):
+    tmp_file = NamedTemporaryFile(dir='/tmp/')
+    with open(tmp_file.name, 'w') as fil:
+        fil.write('\n'.join(node_list))
+    nodes = __nodeattr(file_path=tmp_file.name)('--expand').split("\n")
+    # above split adds trailing empty string in array so
+    del nodes[-1]
+    return nodes
+
+def __nodeattr(file_path=None):
+    if file_path: return local['nodeattr']['-f', file_path]
```

Comment:
In python it should be possible to set a default `file_path` with:
```
def __nodeattr(file_path='/var/lib/adminware/genders'):
    blah
```
This should remove the condition check and the double use of `return`